torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3008,6 +3008,160 @@ struct to_ir {
         return std::make_shared<SimpleValue>(
             emitListComprehension(lc, type_hint));
       }
+      case prim::dict: {
+
+        auto treat_as_empty_container = [&]{
+          if (apply.inputs().empty() && !apply.attributes().empty()) {
+            return true;
+          }
+          if (!apply.inputs().empty() && apply.inputs()[0].kind() == TK_DICT_LITERAL) {
+            auto dict_lit = DictLiteral(apply.inputs()[0]);
+            return dict_lit.key_inputs().empty() && dict_lit.value_inputs().empty();
+          }
+          if (!apply.inputs().empty() && apply.inputs()[0].kind() == TK_LIST_LITERAL) {
+            auto list_lit = ListLiteral(apply.inputs()[0]);
+            return list_lit.inputs().empty();
+          }
+          return false;
+        };
+
+        // If we have a call with an empty container, or if we have a
+        // call with kwargs only
+        if (treat_as_empty_container()) {
+          auto expr_list = List<Expr>::create(apply.range(), {});
+          apply = Apply::create(
+              apply.range(), apply.callee(), expr_list, apply.attributes());
+        }
+
+        // If we have a completely empty call to dict()
+        if (apply.inputs().empty() && apply.attributes().empty()) {
+          TypePtr type = type_hint;
+          if (!type_hint) {
+            type = DictType::create(StringType::get(), TensorType::get());
+          }
+          return std::make_shared<SimpleValue>(
+              graph
+                  ->insertNode(graph->createDict(
+                      type->expect<DictType>()->getKeyType(),
+                      type->expect<DictType>()->getValueType(),
+                      {},
+                      {}))
+                  ->output());
+        }
+
+        // Special case logic for if we have a dict comprehension
+        if (!apply.inputs().empty() && apply.inputs()[0].kind() == TK_DICT_COMP) {
```

Comment:
I think this case would work fine if it was removed. We should try to unify it with the general case below.